src/components/ProductInfo/ProductInfo.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { getProductById } from '../../api/products';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import {
+  setError,
+  setIsLoading,
+  setProduct,
+} from '../../features/productSlice';
+import { Loader } from '../Loader';
+import styles from './ProductInfo.module.scss';
+
+export const ProductInfo: React.FC = () => {
+  const { productId = '' } = useParams();
+  const dispatch = useAppDispatch();
+  const { product, isLoading, error } = useAppSelector(
+    (state) => state.product
+  );
+
+  const propertiesData = [
+    {
+      name: 'Weight (imperial)',
+      value: product?.weight.imperial,
+    },
+    {
+      name: 'Weight (metric)',
+      value: product?.weight.metric,
+    },
+    {
+      name: 'Height (imperial)',
+      value: product?.height.imperial,
+    },
+    {
+      name: 'Height (metric)',
+      value: product?.height.metric,
+    },
+    {
+      name: 'Bred for',
+      value: product?.bred_for,
+    },
+    {
+      name: 'Breed group',
+      value: product?.breed_group,
+    },
+    {
+      name: 'Life span',
+      value: product?.life_span,
+    },
+    {
+      name: 'Temperament',
+      value: product?.temperament,
+    },
+    {
+      name: 'Origin',
+      value: product?.origin,
+    },
+  ];
+
+  const getProductFromServer = async (id: string) => {
+    try {
+      dispatch(setIsLoading(true));
```

Comment:
if you have the logic for storing product, error etc. in redux store it's better to move api request to the redux store as well to have a clear logic of what the store handles and works with

Take a look at https://redux-toolkit.js.org/api/createAsyncThunk